dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/contig/RefseqAssemblyReportParser.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.dbsnpimporter.contig;
+
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.item.file.FlatFileItemReader;
+import org.springframework.batch.item.file.mapping.PassThroughLineMapper;
+import org.springframework.core.io.UrlResource;
+
+import java.net.MalformedURLException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class RefseqAssemblyReportParser {
+
+    private FlatFileItemReader<String> reader;
+
+    private String file;
+
+    private Map<String, String> contigMap;
+
+    public RefseqAssemblyReportParser(String url) {
+        this.file = url;
+        this.contigMap = null;
+        initializeReader(url);
+    }
+
+    private void initializeReader(String url) {
+        reader = new FlatFileItemReader<>();
+        try {
+            reader.setResource(new UrlResource(url));
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException("File location is invalid: " + url, e);
+        }
+        reader.setLineMapper(new PassThroughLineMapper());
+        reader.open(new ExecutionContext());
+    }
+
+    public Map<String, String> getContigMap() throws Exception {
+        if (contigMap == null) {
```

Comment:
I just see this usage:

new RefseqAssemblyReportParser(mappingUrl).getContigMap()

So I think that the null comparision is not needed. Is not harmful either, so this can stay just in case

Reply:
uhm, it is actually lazy evaluation. When the Parser is constructed it opens the file, but doesn't parse it until getContigMap is called, and then, subsequent calls will not parse again the file and will return the already built map.

Reply:
actually, now I think it would be better to initialize the reader here, instead of the constructor.

Reply:
I'm not sure about initializing it on the first run. It is better to catch errors like "File location is invalid" during the construction and preventing the program from progressing.

Reply:
good point. still, I think I should close the file reader in getContigMap